planner/core/preprocess.go

```diff
@@ -485,6 +485,16 @@ func (p *preprocessor) checkCreateTableGrammar(stmt *ast.CreateTableStmt) {
 			}
 		}
 	}
+	for _, option := range stmt.Options {
+		switch option.Tp {
+		case ast.TableOptionUnion:
+			p.err = ddl.ErrTableOptionUnionUnsupported
+			return
+		case ast.TableOptionInsertMethod:
+			p.err = ddl.ErrTableOptionInsertMethodUnsupported
+			return
+		}
+	}
```

Comment:
Is it possible to extract this code into a function？There are two places to use.

Reply:
@zimulala 
fixed in https://github.com/pingcap/tidb/pull/17995/commits/3d7754c49c1e01443fea7012e8202f412a0c6bbf
PTAL, thanks!